### PR TITLE
[AC-1283] AC Team code ownership moves: Organization class

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Move `Organization.cs` (in Entities and EF) into AC Team code ownership.

The organization domain is widely used, so I'm breaking it up into multiple PRs. This will mean more reviews for everyone, but hopefully they'll be tick-and-flick.

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by ike-kottlowski — DISMISSED** [review]

File: src/Admin/Controllers/ToolsController.cs
Comment (djsmith85) [inline src/Admin/Controllers/ToolsController.cs:1]: @cturnbull-bitwarden Should this be owned by @bitwarden/team-billing-dev ?

File: src/Core/Models/Business/OrganizationLicense.cs
Comment (djsmith85) [inline src/Core/Models/Business/OrganizationLicense.cs:1]: @cturnbull-bitwarden: This and anything related to licenses should probably be owned by @bitwarden/team-billing-dev

File: src/Core/Models/Business/SeatSubscriptionUpdate.cs
Comment (djsmith85) [inline src/Core/Models/Business/SeatSubscriptionUpdate.cs:1]: @cturnbull-bitwarden: Anything related to SubscriptionUpdate should be owned by @bitwarden/team-billing-dev

File: src/Core/Services/Implementations/StripePaymentService.cs
Comment (djsmith85) [inline src/Core/Services/Implementations/StripePaymentService.cs:1]: @cturnbull-bitwarden: Should be owned by @bitwarden/team-billing-dev

File: src/Core/Models/Data/Organizations/OrganizationAbility.cs
Comment (djsmith85) [inline src/Core/Models/Data/Organizations/OrganizationAbility.cs:1]: Who should own this? @bitwarden/team-billing-dev or @bitwarden/team-admin-console-dev ?

**Review by djsmith85 — APPROVED** [review]
Changes are looking good.

SCIM related test files should be owned by @bitwarden/team-admin-console-dev

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Pull request merged**